pkg/server/server.go

```diff
@@ -84,6 +85,18 @@ func (s *Server) handleNarinfo(w http.ResponseWriter, r *http.Request) {
 			http.Error(w, fmt.Sprintf("handle-narinfo: %v", err), http.StatusBadRequest)
 			return
 		}
+
+		// rewrite narinfo, as we undo all compression and serve them from /nar/$narhash.nar
+		ni.Compression = "none"
+		ni.FileHash = ni.NarHash
+		ni.FileSize = ni.NarSize
+		ni.URL = "nar/" + nixbase32.EncodeToString(ni.NarHash.Digest) + ".nar"
+
+		// TODO: can this ever happen?
```

Comment:
I'm not sure about FOD, does it keep the original hash or calculate its own?

Reply:
I'm worried about Nix specifying hashes different than sha256.

As of https://github.com/flokli/nix-casync/commit/ed5fe4c271a2633bdd539ad4e107479d41a8a67b, the `sha256` is what we use to address `.nar` files. If `NarHash` in the uploaded narfile is not `sha256`, we don't know how to lookup that Narfile.

This could be solved too if we would calculate all hash types, but then we'd need a little database to lookup the sha256 of a file with sha1 X.

For now, this rejects the upload with a "Bad Request" if a hash != sha256 is used - not sure if that's a problem.